### ISSUE TITLE
non-virtual member function marked 'override' hides virtual member function

### ISSUE THREAD

**Issue by AA1999** [llvm/llvm-project#61116]
While compiling llvm:

```cpp
[303/8867] Building CXX object lib/FuzzMutate/CMakeFiles/LLVMFuzzerCLI.dir/FuzzerCLI.cpp.o
[4762/8861] Building LLDB Lua wrapper
/home/arshia/Desktop/llvm-project/lldb/include/lldb/API/SBCommandReturnObject.h:103: Warning 509: Overloaded method lldb::SBCommandReturnObject::PutCString(char const *) effectively ignored,
/home/arshia/Desktop/llvm-project/lldb/include/lldb/API/SBCommandReturnObject.h:103: Warning 509: as it is shadowed by lldb::SBCommandReturnObject::PutCString(char const *,int).
[4765/8861] Building LLDB Python wrapper
/home/arshia/Desktop/llvm-project/lldb/include/lldb/API/SBCommandReturnObject.h:103: Warning 509: Overloaded method lldb::SBCommandReturnObject::PutCString(char const *) effectively ignored,
/home/arshia/Desktop/llvm-project/lldb/include/lldb/API/SBCommandReturnObject.h:103: Warning 509: as it is shadowed by lldb::SBCommandReturnObject::PutCString(char const *,int).
[5351/8861] Building CXX object tools/lldb/source/Plugins/ScriptInterpreter/Lua/CMakeFiles/lldbPluginScriptInterpreterLua.dir/ScriptInterpreterLua.cpp.o
FAILED: tools/lldb/source/Plugins/ScriptInterpreter/Lua/CMakeFiles/lldbPluginScriptInterpreterLua.dir/ScriptInterpreterLua.cpp.o 
/usr/bin/clang++ -DGTEST_HAS_RTTI=0 -DHAVE_ROUND -DLLDB_CONFIGURATION_DEBUG -D_DEBUG -D_GLIBCXX_ASSERTIONS -D_GNU_SOURCE -D_LIBCPP_ENABLE_ASSERTIONS -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/home/arshia/Desktop/llvm-project/build/tools/lldb/source/Plugins/ScriptInterpreter/Lua -I/home/arshia/Desktop/llvm-project/lldb/source/Plugins/ScriptInterpreter/Lua -I/home/arshia/Desktop/llvm-project/lldb/include -I/home/arshia/Desktop/llvm-project/build/tools/lldb/include -I/home/arshia/Desktop/llvm-project/build/include -I/home/arshia/Desktop/llvm-project/llvm/include -I/usr/include/python3.10 -I/home/arshia/Desktop/llvm-project/llvm/../clang/include -I/home/arshia/Desktop/llvm-project/build/tools/lldb/../clang/include -I/home/arshia/Desktop/llvm-project/lldb/source -I/home/arshia/Desktop/llvm-project/build/tools/lldb/source -I/usr/include/lua5.3 -isystem /usr/include/libxml2 -O3 -pipe -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -Wno-deprecated-declarations -Wno-unknown-pragmas -Wno-strict-aliasing -Wno-deprecated-register -Wno-vla-extension -g  -fno-exceptions -fno-rtti -std=c++17 -MD -MT tools/lldb/source/Plugins/ScriptInterpreter/Lua/CMakeFiles/lldbPluginScriptInterpreterLua.dir/ScriptInterpreterLua.cpp.o -MF tools/lldb/source/Plugins/ScriptInterpreter/Lua/CMakeFiles/lldbPluginScriptInterpreterLua.dir/ScriptInterpreterLua.cpp.o.d -o tools/lldb/source/Plugins/ScriptInterpreter/Lua/CMakeFiles/lldbPluginScriptInterpreterLua.dir/ScriptInterpreterLua.cpp.o -c /home/arshia/Desktop/llvm-project/lldb/source/Plugins/ScriptInterpreter/Lua/ScriptInterpreterLua.cpp
In file included from /home/arshia/Desktop/llvm-project/lldb/source/Plugins/ScriptInterpreter/Lua/ScriptInterpreterLua.cpp:9:
/home/arshia/Desktop/llvm-project/lldb/source/Plugins/ScriptInterpreter/Lua/ScriptInterpreterLua.h:95:68: error: non-virtual member function marked 'override' hides virtual member function
                                    const char *command_body_text) override;
                                                                   ^
/home/arshia/Desktop/llvm-project/lldb/include/lldb/Interpreter/ScriptInterpreter.h:415:16: note: hidden overloaded virtual function 'lldb_private::ScriptInterpreter::SetWatchpointCommandCallback' declared here: different number of parameters (3 vs 2)
  virtual void SetWatchpointCommandCallback(WatchpointOptions *wp_options,
               ^
/home/arshia/Desktop/llvm-project/lldb/source/Plugins/ScriptInterpreter/Lua/ScriptInterpreterLua.cpp:114:37: error: too few arguments to function call, expected 3, have 2
            bp_options, data.c_str());
                                    ^
/home/arshia/Desktop/llvm-project/lldb/source/Plugins/ScriptInterpreter/Lua/ScriptInterpreterLua.h:90:10: note: 'SetBreakpointCommandCallback' declared here
  Status SetBreakpointCommandCallback(BreakpointOptions &bp_options,
         ^
/home/arshia/Desktop/llvm-project/lldb/source/Plugins/ScriptInterpreter/Lua/ScriptInterpreterLua.cpp:350:30: error: out-of-line definition of 'SetBreakpointCommandCallback' does not match any declaration in 'lldb_private::ScriptInterpreterLua'
Status ScriptInterpreterLua::SetBreakpointCommandCallback(
                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
3 errors generated.
[5354/8861] Building CXX object tools/lldb/source/Plugins/SymbolFile/Breakpad/CMakeFiles/lldbPluginSymbolFileBreakpad.dir/SymbolFileBreakpad.cpp.o
ninja: build stopped: subcommand failed.
[408/4058] cd /home/arshia/Desktop/llvm-project/clang/bindings/python && /usr/...arshia/Desktop/llvm-project/build/lib /usr/bin/python3.10 -m unittest discover
................................................................................................................................
----------------------------------------------------------------------
Ran 128 tests in 2.135s

OK
[411/4058] Building CXX object tools/lldb/source/Plugins/ScriptInterpreter/Lua/CMakeFiles/lldbPluginScriptInterpreterLua.dir/ScriptInterpreterLua.cpp.o
FAILED: tools/lldb/source/Plugins/ScriptInterpreter/Lua/CMakeFiles/lldbPluginScriptInterpreterLua.dir/ScriptInterpreterLua.cpp.o 
/usr/bin/clang++ -DGTEST_HAS_RTTI=0 -DHAVE_ROUND -DLLDB_CONFIGURATION_DEBUG -D_DEBUG -D_GLIBCXX_ASSERTIONS -D_GNU_SOURCE -D_LIBCPP_ENABLE_ASSERTIONS -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/home/arshia/Desktop/llvm-project/build/tools/lldb/source/Plugins/ScriptInterpreter/Lua -I/home/arshia/Desktop/llvm-project/lldb/source/Plugins/ScriptInterpreter/Lua -I/home/arshia/Desktop/llvm-project/lldb/include -I/home/arshia/Desktop/llvm-project/build/tools/lldb/include -I/home/arshia/Desktop/llvm-project/build/include -I/home/arshia/Desktop/llvm-project/llvm/include -I/usr/include/python3.10 -I/home/arshia/Desktop/llvm-project/llvm/../clang/include -I/home/arshia/Desktop/llvm-project/build/tools/lldb/../clang/include -I/home/arshia/Desktop/llvm-project/lldb/source -I/home/arshia/Desktop/llvm-project/build/tools/lldb/source -I/usr/include/lua5.3 -isystem /usr/include/libxml2 -O3 -pipe -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -Wno-deprecated-declarations -Wno-unknown-pragmas -Wno-strict-aliasing -Wno-deprecated-register -Wno-vla-extension -g  -fno-exceptions -fno-rtti -std=c++17 -MD -MT tools/lldb/source/Plugins/ScriptInterpreter/Lua/CMakeFiles/lldbPluginScriptInterpreterLua.dir/ScriptInterpreterLua.cpp.o -MF tools/lldb/source/Plugins/ScriptInterpreter/Lua/CMakeFiles/lldbPluginSc
╭─[arshia-pc] as arshia in ~/Desktop/llvm-project on (main)✘✘✘                                                                                         12:21:07
╰─(ﾉ˚Д˚)ﾉ cmake --build build -j4   
[4/3510] Building CXX object tools/lldb/source/Plugins/ScriptInterpreter/Lua/CMakeFiles/lldbPluginScriptInterpreterLua.dir/ScriptInterpreterLua.cpp.o
FAILED: tools/lldb/source/Plugins/ScriptInterpreter/Lua/CMakeFiles/lldbPluginScriptInterpreterLua.dir/ScriptInterpreterLua.cpp.o 
/usr/bin/clang++ -DGTEST_HAS_RTTI=0 -DHAVE_ROUND -DLLDB_CONFIGURATION_DEBUG -D_DEBUG -D_GLIBCXX_ASSERTIONS -D_GNU_SOURCE -D_LIBCPP_ENABLE_ASSERTIONS -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/home/arshia/Desktop/llvm-project/build/tools/lldb/source/Plugins/ScriptInterpreter/Lua -I/home/arshia/Desktop/llvm-project/lldb/source/Plugins/ScriptInterpreter/Lua -I/home/arshia/Desktop/llvm-project/lldb/include -I/home/arshia/Desktop/llvm-project/build/tools/lldb/include -I/home/arshia/Desktop/llvm-project/build/include -I/home/arshia/Desktop/llvm-project/llvm/include -I/usr/include/python3.10 -I/home/arshia/Desktop/llvm-project/llvm/../clang/include -I/home/arshia/Desktop/llvm-project/build/tools/lldb/../clang/include -I/home/arshia/Desktop/llvm-project/lldb/source -I/home/arshia/Desktop/llvm-project/build/tools/lldb/source -I/usr/include/lua5.3 -isystem /usr/include/libxml2 -O3 -pipe -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -Wno-deprecated-declarations -Wno-unknown-pragmas -Wno-strict-aliasing -Wno-deprecated-register -Wno-vla-extension -g  -fno-exceptions -fno-rtti -std=c++17 -MD -MT tools/lldb/source/Plugins/ScriptInterpreter/Lua/CMakeFiles/lldbPluginScriptInterpreterLua.dir/ScriptInterpreterLua.cpp.o -MF tools/lldb/source/Plugins/ScriptInterpreter/Lua/CMakeFiles/lldbPluginScriptInterpreterLua.dir/ScriptInterpreterLua.cpp.o.d -o tools/lldb/source/Plugins/ScriptInterpreter/Lua/CMakeFiles/lldbPluginScriptInterpreterLua.dir/ScriptInterpreterLua.cpp.o -c /home/arshia/Desktop/llvm-project/lldb/source/Plugins/ScriptInterpreter/Lua/ScriptInterpreterLua.cpp
In file included from /home/arshia/Desktop/llvm-project/lldb/source/Plugins/ScriptInterpreter/Lua/ScriptInterpreterLua.cpp:9:
/home/arshia/Desktop/llvm-project/lldb/source/Plugins/ScriptInterpreter/Lua/ScriptInterpreterLua.h:95:68: error: non-virtual member function marked 'override' hides virtual member function
                                    const char *command_body_text) override;
                                                                   ^
/home/arshia/Desktop/llvm-project/lldb/include/lldb/Interpreter/ScriptInterpreter.h:415:16: note: hidden overloaded virtual function 'lldb_private::ScriptInterpreter::SetWatchpointCommandCallback' declared here: different number of parameters (3 vs 2)
  virtual void SetWatchpointCommandCallback(WatchpointOptions *wp_options,
               ^
/home/arshia/Desktop/llvm-project/lldb/source/Plugins/ScriptInterpreter/Lua/ScriptInterpreterLua.cpp:114:37: error: too few arguments to function call, expected 3, have 2
            bp_options, data.c_str());
                                    ^
/home/arshia/Desktop/llvm-project/lldb/source/Plugins/ScriptInterpreter/Lua/ScriptInterpreterLua.h:90:10: note: 'SetBreakpointCommandCallback' declared here
  Status SetBreakpointCommandCallback(BreakpointOptions &bp_options,
         ^
/home/arshia/Desktop/llvm-project/lldb/source/Plugins/ScriptInterpreter/Lua/ScriptInterpreterLua.cpp:350:30: error: out-of-line definition of 'SetBreakpointCommandCallback' does not match any declaration in 'lldb_private::ScriptInterpreterLua'
Status ScriptInterpreterLua::SetBreakpointCommandCallback(
                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
3 errors generated.
[7/3510] Building CXX object tools/lldb/source/Plugins/SymbolFile/DWARF/CMakeFiles/lldbPluginSymbolFileDWARF.dir/DWARFASTParser.cpp.o
ninja: build stopped: subcommand failed.   
```

**Comment by shafik** [comment]
CC @JDevlieghere @medismailben

**Comment by medismailben** [comment]
@shafik should be fixed with ab81fc29d9f7
